File: DontStarve/BarsInformation.cs
using DontStarve.Player.Hunger;
using DontStarve.Player.Sanity;
using Microsoft.Xna.Framework;
using StardewValley;

namespace DontStarve;

internal static class BarsInformation
{
    private static readonly Color fullSanityColor = new(0xFF, 0xC7, 0x00);
    private static readonly Color zeroSanityColor = new(0xA9, 0xA9, 0xA9);
    private static readonly Color fullHungerColor = new(0xFF, 0xC7, 0x00);
    private static readonly Color zeroHungerColor = new(0xA9, 0xA9, 0xA9);

    internal static Color sanityColor
    {
        get
        {
            var player = Game1.player;
            var percent = player.getSanity() / player.getMaxSanity();
            var lerpR = fullSanityColor.R - zeroSanityColor.R;
            var lerpG = fullSanityColor.G - zeroSanityColor.G;
            var lerpB = fullSanityColor.B - zeroSanityColor.B;
            return new Color(
                zeroSanityColor.R + (int)(percent * lerpR),
                zeroSanityColor.G + (int)(percent * lerpG),
                zeroSanityColor.B + (int)(percent * lerpB)
            );
        }
    }

    internal static Color hungerColor
    {
        get
        {
            var player = Game1.player;
            var percent = player.getHunger() / player.getMaxHunger();
            var lerpR = fullHungerColor.R - zeroHungerColor.R;
            var lerpG = fullHungerColor.G - zeroHungerColor.G;
            var lerpB = fullHungerColor.B - zeroHungerColor.B;
            return new Color(
                zeroHungerColor.R + (int)(percent * lerpR),
                zeroHungerColor.G + (int)(percent * lerpG),
                zeroHungerColor.B + (int)(percent * lerpB)
            );
        }
    }
}

File: kotlin/src/nativeMain/kotlin/BarsInformation.kt
import Hunger.getHunger
import Hunger.getMaxHunger
import Sanity.getMaxSanity
import Sanity.getSanity
import wrapper.Color
import wrapper.Game1

object BarsInformation {
	private val fullSanityColor = Color(0xFF, 0xC7, 0x00)
	private val zeroSanityColor = Color(0xA9, 0xA9, 0xA9)
	private val fullHungerColor = Color(0xFF, 0xC7, 0x00)
	private val zeroHungerColor = Color(0xA9, 0xA9, 0xA9)

	val sanityColor: Color
		get() {
			val player = Game1.player
			val percent = player.getSanity() / player.getMaxSanity()
			val lerpR = fullSanityColor.r - zeroSanityColor.r
			val lerpG = fullSanityColor.g - zeroSanityColor.g
			val lerpB = fullSanityColor.b - zeroSanityColor.b
			return Color(
				zeroSanityColor.r + (percent * lerpR).toInt(),
				zeroSanityColor.r + (percent * lerpG).toInt(),
				zeroSanityColor.r + (percent * lerpB).toInt()
			)
		}

	val hungerColor: Color
		get() {
			val player = Game1.player
			val percent = player.getHunger() / player.getMaxHunger()
			val lerpR = fullHungerColor.r - zeroHungerColor.r
			val lerpG = fullHungerColor.g - zeroHungerColor.g
			val lerpB = fullHungerColor.b - zeroHungerColor.b
			return Color(
				zeroHungerColor.r + (percent * lerpR).toInt(),
				zeroHungerColor.g + (percent * lerpG).toInt(),
				zeroHungerColor.b + (percent * lerpB).toInt()
			)
		}
}